pallets/msa/src/tests/signed_extension_tests.rs

```diff
@@ -507,18 +508,45 @@ fn signed_ext_validate_fails_when_withdraw_tokens_caller_key_does_not_match_payl
 	});
 }
 
+#[test]
+fn signed_ext_validate_fails_when_withdraw_tokens_payload_payload_is_has_wrong_type_discriminant() {
```

Comment:
nit: this sounds funny to me
```suggestion
fn signed_ext_validate_fails_when_withdraw_tokens_payload_has_wrong_type_discriminant() {
```